src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2632,16 +2650,175 @@ CalculateNewConnectionCount(WorkerPool *workerPool)
 		 * than the target pool size.
 		 */
 		newConnectionCount = Min(newConnectionsForReadyTasks, maxNewConnectionCount);
-		if (newConnectionCount > 0)
+		if (initiatedConnectionCount <= MaxCachedConnectionsPerWorker)
```

Comment:
I added this check just now, in case you haven't seen @marcocitus

Reply:
it's not entirely clear to me why this is separate from the 3 other conditions, which makes it a bit confusing to read. 

Maybe separate out the conditions or have 1 comment describing all of them.

Reply:
my intention was to simplify readability, but if having multiple checks confused you, then it is probably better idea to have a single check :)